dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java

```diff
@@ -531,6 +531,7 @@ private boolean checkResourceExists(String fullName, int userId, int type) {
         List<Integer> resourcesIds = resourceUserMapper.queryResourcesIdListByUserIdAndPerm(userId, 0);
 
         IPage<Resource> resourceIPage = resourcesMapper.queryResourcePaging(page, userId, directoryId, type.ordinal(), searchVal,resourcesIds);
+        resourceIPage.getRecords().forEach(resource -> resource.setUserName(loginUser.getUserName()));
```

Comment:
This does not seem to be a good implementation.
There may be a case that the login user is admin user, so he will see all resource's belong user is himself.